samples/deploy_ova.py

```diff
@@ -264,8 +275,19 @@ def upload_disk(self, file_item, lease, host):
             ssl_context = ssl._create_unverified_context()
         else:
             ssl_context = None
-        req = Request(url, ovffile, headers)
-        urlopen(req, context=ssl_context)
+
+        method = 'POST'
+        if not device_url.disk:
+            headers['Overwrite'] = 't'
+            method = 'PUT'
+        
+        p = urlparse(url)
+
+        conn = http.client.HTTPSConnection(p.netloc, context=ssl_context,
```

Comment:
I'm not an expert here, but if the url provided is not https, would this still work? (not sure how many people still use plain http, but might be used on internal sites)

Reply:
All the clusters I've worked with were always using https, (even if with self-signed certs most of the time), but I think it's a fair point. Let me see if I can come up with a fix that takes it into consideration.

Reply:
Well in fact it turns out that, just like `urlopen`,  `HTTPSConnection` works with HTTP urls as well, as demonstrated by the following snippet:

```
import http.client
import ssl
from urllib.parse import urlparse

url1 = 'http://httpbin.org/'
url2 = 'https://httpbin.org/'
p1 = urlparse(url1)
p2 = urlparse(url2)

ssl_context = ssl._create_unverified_context()

conn1 = http.client.HTTPSConnection(p1.netloc, context=ssl_context)
conn2 = http.client.HTTPSConnection(p2.netloc, context=ssl_context)
conn1.request('GET', p1.path)
conn2.request('GET', p2.path)

r1 = conn1.getresponse()
r2 = conn2.getresponse()

print(r1.read(10))
print(r2.read(10))
```

output:

```
b'<!DOCTYPE '
b'<!DOCTYPE '
```

so the code should already be safe as is.

Reply:
No, wrong, sorry, my test case was all wrong. Let me fix it.